tsl/src/continuous_aggs/materialize.c

```diff
@@ -1163,6 +1205,29 @@ range_expand_right(InternalTimeRange *range, const InternalTimeRange to_cover, i
 	return false;
 }
 
+static void
+range_clamp_end(InternalTimeRange *range, int64 end_value, int64 bucket_width)
+{
+	if (end_value == PG_INT64_MAX)
+		return;
+
+	/* the end value has to be bucketed, we don't bucket here since sometimes you want to round down
+	 * or round up and the caller should decide. */
+	Assert(end_value == ts_time_bucket_by_type(bucket_width, end_value, range->type));
+	if (end_value < range->start)
```

Comment:
I think this should be an Assertion failure, it violates the invariant of the range type

Reply:
whoops, misread the condition. Considering what this actually does I think the body should be
```
range->start = end;
range->end = end;
```
but not set on it